### ISSUE TITLE
[linky] Full history of data returned by the requests

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab-addons#17070]
Start and end dates in the console command are ignored. Even default dates are ignored. Looks like around 3 years of data are systematically returned.
Maybe something has changed in the API with the handling of dates ?
If only thie "report" console command is affected, that is not dramatic. In case all daily requests are affected, that would be more annoying.
I will investigate.
@clinique for information

**Comment by lolodomo** [comment]
Just bu enabling DEBUG logs, I can unfortunately confirm that even daily requests are impacted, that means each request returns a full history. I guess the dates you provide are probably wrong, maybe the format has changed ?

**Comment by lolodomo** [comment]
@clinique : any chance you take a look ? Such a fix could be a candidate for 4.2.x.

**Comment by clinique** [comment]
Morning @lolodomo . I've made the check in the browser.
The URL parameters does not seem to have changed: 
![image](https://github.com/user-attachments/assets/d3dd011a-d742-4519-a273-acf452089ccd)

...just they seem to be completely ignored by the server : 
![image](https://github.com/user-attachments/assets/44d3ea20-aeae-47a5-bb5b-0e786a993210)

**Comment by lolodomo** [comment]
What do you suggest?

**Comment by clinique** [comment]
I don't know what I can do. The API does not take care of date filtering

**Comment by lsiepel** [comment]
Is there official documentation for this api?
I found https://data.enedis.fr/api/v1/console/datasets/1.0/search/ but that might be totally unrelated. If it is the same api, there is also a v2 and newer api. Those might not be affected (long shot)

**Comment by clinique** [comment]
No, it has be retro-engineered from the website. Final solution should come from https://github.com/openhab/openhab-addons/pull/16355

**Comment by clinique** [comment]
Nothing doable on Linky binding side.

**Comment by lolodomo** [comment]
In case we have the full history systematically, we have to redesign the report console command.

**Comment by lolodomo** [comment]
@lo92fr : can you confirm that now the API systematically returns the full history of day values (2 years - or 3) ?
If true, I will have to simplify the code generating the report run from the console command.

**Comment by lo92fr** [comment]
Hello @lolodomo,

I confirm, on the api side, we currently request the full data (3 years) on each call.
So you can simplify the code if you like.

Laurent.
